### PR TITLE
Add delete method

### PR DESCRIPTION
**Pull request by yaromeister**
Refactor tests, refactor validators, changed IUserService to take params

### CODE REVIEW

File: VetClinic.BLL.Tests/AutoMoqDataAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        public AutoMoqDataAttribute() : base(() => {
+            var fixture = new Fixture().Customize(new AutoMoqCustomization());
+
+            fixture.Behaviors.OfType<ThrowingRecursionBehavior>().ToList().ForEach(b => fixture.Behaviors.Remove(b));
+            fixture.Behaviors.Add(new OmitOnRecursionBehavior());
+
+            return fixture;
+        })
+        {
+
```
Comment (Serafinium) [inline VetClinic.BLL.Tests/AutoMoqDataAttribute.cs:19]: Why do you need the current block?

File: VetClinic.BLL.Tests/Services/UserServiceTests.cs [hunk 2]
```diff
@@ -12,100 +10,139 @@ namespace VetClinic.BLL.Tests.Services
...
-
-        [Fact]
-        public async Task CreateUser_UserExsists_ReturnFalse()
+        [Theory, AutoMoqData]
+        public async Task CreateUser_UserExsists_ReturnFalse(User user,
+            IdentityRole[] roles,
+            RoleManager<IdentityRole> roleManager,
+            [Frozen] Mock<UserManager<User>> userManager)
         {
             //arrange
```
Comment (Serafinium) [inline VetClinic.BLL.Tests/Services/UserServiceTests.cs:19]: Please use upper case

File: VetClinic.BLL.Tests/Services/UserServiceTests.cs [hunk 3]
```diff
@@ -12,100 +10,139 @@ namespace VetClinic.BLL.Tests.Services
...
-        [Fact]
-        public async Task CreateUser_CreateFailure_ReturnFalse()
+        
+        [Theory, AutoMoqData]
+        public async Task CreateUser_CreateFailure_ReturnFalse(User user,
+            IdentityRole[] roles,
+            RoleManager<IdentityRole> roleManager,
+            [Frozen] Mock<UserManager<User>> userManager)
         {
             //arrange
```
Comment (Serafinium) [inline VetClinic.BLL.Tests/Services/UserServiceTests.cs:39]: The same above

File: VetClinic.BLL.Tests/Services/UserServiceTests.cs [hunk 4]
```diff
@@ -12,100 +10,139 @@ namespace VetClinic.BLL.Tests.Services
...
-        [Fact]
-        public async Task CreateUser_UserExsists_ReturnFalse()
+        [Theory, AutoMoqData]
+        public async Task CreateUser_UserExsists_ReturnFalse(User user,
+            IdentityRole[] roles,
+            RoleManager<IdentityRole> roleManager,
+            [Frozen] Mock<UserManager<User>> userManager)
         {
             //arrange
-            var mgr = new Mock<UserManager<User>>(UserStore.Object, null, null, null, null, null, null, null, null);
```
Comment (Serafinium) [inline VetClinic.BLL.Tests/Services/UserServiceTests.cs:32]: mgr -  what does it means? Please use more meaningful name.

File: VetClinic.BLL.Tests/Services/UserServiceTests.cs [hunk 5]
```diff
@@ -12,100 +10,139 @@ namespace VetClinic.BLL.Tests.Services
...
-        public async Task CreateUser_CreateFailure_ReturnFalse()
+        
+        [Theory, AutoMoqData]
+        public async Task CreateUser_CreateFailure_ReturnFalse(User user,
+            IdentityRole[] roles,
+            RoleManager<IdentityRole> roleManager,
+            [Frozen] Mock<UserManager<User>> userManager)
         {
             //arrange
-            var mgr = new Mock<UserManager<User>>(UserStore.Object, null, null, null, null, null, null, null, null);
```
Comment (Serafinium) [inline VetClinic.BLL.Tests/Services/UserServiceTests.cs:50]: mgr -  what does it means? Please use more meaningful name.

File: VetClinic.BLL.Tests/Services/UserServiceTests.cs [hunk 6]
```diff
@@ -12,100 +10,139 @@ namespace VetClinic.BLL.Tests.Services
...
-        [Fact]
-        public async Task CreateUser_CreateSuccess_ReturnTrue()
+        [Theory, AutoMoqData]
+        public async Task CreateUser_CreateSuccess_ReturnTrue(User user,
+            IdentityRole roles,
+            [Frozen] Mock<RoleManager<IdentityRole>> roleManager,
+            [Frozen] Mock<UserManager<User>> userManager)
         {
             //arrange
-            var mgr = new Mock<UserManager<User>>(UserStore.Object, null, null, null, null, null, null, null, null);
```
Comment (Serafinium) [inline VetClinic.BLL.Tests/Services/UserServiceTests.cs:72]: Please use more meaningful name.

File: VetClinic.BLL.Tests/AutoMoqDataAttribute.cs (hunk 1 above)
Comment (yaromeister) [reply VetClinic.BLL.Tests/AutoMoqDataAttribute.cs:19]: It's just an empty costructor, all the magic happens in the base(...). Or do you mean that i should return it inline?

File: VetClinic.BLL/Services/Realizations/UserService.cs [hunk 7]
```diff
@@ -56,7 +56,7 @@ public async Task<bool> UpdateUser(string id, User inputUser, IEnumerable<Identi
                 user.LastName = inputUser.LastName;
                 user.Email = inputUser.Email;
                 user.PhoneNumber = inputUser.PhoneNumber;
-
+
```
Comment (Serafinium) [inline VetClinic.BLL/Services/Realizations/UserService.cs:59]: Please align code

File: VetClinic.API/Validators/User/CreateUserDtoValidator.cs [hunk 8]
```diff
@@ -7,20 +8,26 @@ public class CreateUserDtoValidator : AbstractValidator<CreateUserDto>
...
-            RuleFor(user => user.Email).NotEmpty().WithMessage("Email не може бути порожнім");
-            RuleFor(user => user.Email).EmailAddress().WithMessage("Email невірного формату");
-            RuleFor(user => user.Email).MaximumLength(50).WithMessage("Email не може бути довше 50 символів");
-            RuleFor(user => user.PhoneNumber).NotEmpty().WithMessage("Телефон не може бути порожнім");
-            RuleFor(user => user.PhoneNumber).MaximumLength(12).WithMessage("Телефон не може бути довше 12 символів");
-            RuleFor(user => user.Password).NotEmpty().WithMessage("Пароль не може бути порожнім");
-            RuleFor(user => user.Password).MaximumLength(300).WithMessage("Пароль занадто довгий");
-            RuleFor(user => user.Password).MinimumLength(6).WithMessage("Пароль занадто короткий");
+            RuleFor(user => user.UserName).NotEmpty().WithMessage("Нікнейм не може бути порожнім")
+                .MaximumLength(50).WithMessage("Нікнейм не може бути довшим 50 символів");
```
Comment (Serafinium) [inline VetClinic.API/Validators/User/CreateUserDtoValidator.cs:12]: Let's use English explanation. Please use English meassages

File: VetClinic.BLL/Services/Realizations/UserService.cs (hunk 7 above)
Comment (yaromeister) [reply VetClinic.BLL/Services/Realizations/UserService.cs:59]: Isn't it already aligned? It does have one tab after if block brace

**Pull request merged**